src/server.rs

```diff
@@ -101,9 +99,93 @@ pub enum Event {
     },
 }
 
+/// Data structure containing data to be sent by server in messages containing rectangles.
+#[derive(Debug)]
+pub enum Rectangle<'a> {
```

Comment:
A more fitting name for this would be an `Update`, since half of the enum variants have nothing to do with rectangles at all.